src/java.desktop/share/classes/javax/swing/JLabel.java

```diff
@@ -1089,6 +1095,25 @@ public String getAccessibleName() {
             return name;
         }
 
+        private String getAccessibleNameCheckIcon(String name) {
+            // If an icon is set but no text is set,
+            // the screen reader will say "image".
+            // cc jdk-8277497
+            if (((name == null) || name.isEmpty()) &&
+                    (JLabel.this.getIcon() != null)) {
+                if (JLabel.this.getIcon() instanceof Accessible) {
+                    AccessibleContext ac = ((Accessible) JLabel.this.getIcon()).getAccessibleContext();
+                    if (ac != null) {
+                        name = ac.getAccessibleName();
+                    }
+                }
+                if ((name == null) || name.isEmpty()) {
```

Comment:
I continue our discussion from code line above.

Why for the empty label and non-accessible image we should say "image", and not for example the "empty label" or just ignore it? 

I mean that if the icon used by the label does not have an "accessible" icon, or any other accessible component why we will try to mention it to the user?

If non-accessible component should not be mentioned to user then you do not need to add a new code here, we already have the "getAccessibleIcon" method which can be called for example from the "CAccessibility" every time we request name/description/text and it empty while the icon is not null and accessible - this will cover all  accesible components not only JLabel.

BTW it could a SwingSet/App bug that the provided image does not have a description (similar to the common bug in html when the image tag does not have an alt).

Reply:
Looks like If the change will be done in the "CAccessibility" it will cover the custom components as well.

Reply:
Thank you. We’ve considered your suggestion and found it reasonable. Indeed, it should be enough here to propagate the icon’s accessible name when it exists. Also, we will change SwingSet2 so that those icons had accessible names.